cassandra-dac/src/main/java/org/sunbird/helper/CassandraConnectionManagerImpl.java

```diff
@@ -0,0 +1,270 @@
+        }
+        final Metadata metadata = cluster.getMetadata();
+        String msg = String.format("Connected to cluster: %s", metadata.getClusterName());
+        ProjectLogger.log(msg);
+
+        for (final Host host : metadata.getAllHosts()) {
+          msg = String
+              .format("Datacenter: %s; Host: %s; Rack: %s", host.getDatacenter(), host.getAddress(),
+                  host.getRack());
+          ProjectLogger.log(msg);
+        }
+      }
+    } catch (Exception e) {
+      ProjectLogger.log("Error occured while creating connection :", e);
+      throw new ProjectCommonException(ResponseCode.internalError.getErrorCode(), e.getMessage(),
+          ResponseCode.SERVER_ERROR.getResponseCode());
+    }
+
+    if (null != cassandraSessionMap.get(keyspace)) {
+      connection = true;
+    }
+    return connection;
+
+
+  }
+
+  /**
+   * @param ip String
+   * @param port String
+   * @param userName String
+   * @param password String
+   * @param poolingOptions PoolingOptions
+   * @return Cluster Cluster
+   */
+  private static Cluster createCluster(String ip, String port, String userName, String password,
+      PoolingOptions poolingOptions) {
+    return Cluster
+        .builder()
+        .addContactPoint(ip)
+        .withPort(Integer.parseInt(port))
+        .withProtocolVersion(ProtocolVersion.V3)
+        .withRetryPolicy(DefaultRetryPolicy.INSTANCE)
+        .withTimestampGenerator(new AtomicMonotonicTimestampGenerator())
+        .withPoolingOptions(poolingOptions).withCredentials(userName, password).build();
+  }
+
+  /**
+   * @param ip String
+   * @param port String
+   * @param poolingOptions PoolingOptions
+   * @return Cluster Cluster
+   */
+  private static Cluster createCluster(String ip, String port, PoolingOptions poolingOptions) {
+    return Cluster
+        .builder()
+        .addContactPoint(ip)
+        .withPort(Integer.parseInt(port))
+        .withProtocolVersion(ProtocolVersion.V3)
+        .withRetryPolicy(DefaultRetryPolicy.INSTANCE)
+        .withTimestampGenerator(new AtomicMonotonicTimestampGenerator())
+        .withPoolingOptions(poolingOptions).build();
+  }
+
+  /**
+   * Method to create the embedded cassandra connection on localhost and port 9142 by default.
+   * @param ip
+   * @param port
+   * @param userName
+   * @param password
+   * @param keyspace
+   * @return
+   */
+  private boolean createEmbeddedConnection(String ip, String port, String userName, String password,
+      String keyspace) {
+
+    Session cassandraSession = null;
+
+    PropertiesCache propertiesCache = PropertiesCache.getInstance();
+    boolean connection = false;
+
+    if (null == cassandraSessionMap.get(keyspace)) {
+      try {
+        EmbeddedCassandraServerHelper.startEmbeddedCassandra(Long.parseLong(propertiesCache.getProperty("embeddedCassandra_TimeOut")));
+        Cluster cluster = new Cluster.Builder().addContactPoints("127.0.0.1").withPort(9142)
```

Comment:
Put ip 127.0.0.1  and port  : 9142  into external resource file or cassandra resource file.